javalib/src/main/scala/java/net/PlainSocketImpl.scala

```diff
@@ -125,7 +125,7 @@ private[net] class PlainSocketImpl extends SocketImpl {
 
       pollFdPtr.fd = fd.fd
       pollFdPtr.events = POLLIN
-      pollFdPtr.revents = 0
```

Comment:
I think these probably should be removed complete or left as is even though it may not be needed especially if that is idiomatic C code. In `SocketHelpers` you removed the `= 0` and `= null` so either way it should be consistent.

Reply:
Eric,

Please help me understand.   When I view the `PlainSocketImpl.scala` in this PR, at the lines you
indicate I get:
```
 pollFdPtr.fd = fd.fd
 pollFdPtr.events = POLLIN
 // pollFdPtr.revents already zero, stackalloc cleared.
```
Is it the one line comment that you do not like?

That line is a kindness & gift to present & future readers.

Any person trying to trace that code is bound to ask: what is the value of revents? 
Especially since revents is reset later in the method.  

The one line comment succinctly answers that question and also gives a clue
about the astonishing stackalloc behavior.  If it saves even one person
time and good will, it is worth it.

Without the comment and until the unsafe.package stackalloc description
and the ReadTheDocs .rst can be corrected, anyone reading that section
will believe that there is a bug (that is, revents is not initialized).

I did something conceptually identical (consistent) but textually different in
SocketHelpers.scala.

old:
```
 hints.ai_family = AF_UNSPEC
      hints.ai_protocol = 0
      hints.ai_addr = null
      hints.ai_flags = 4 // AI_NUMERICHOST
      hints.ai_socktype = SOCK_STREAM
      hints.ai_next = null
```

new:
```
      val hints = alloc[addrinfo] // alloc documented to clear retured memory 
      hints.ai_family = AF_UNSPEC
      hints.ai_flags = 4 // AI_NUMERICHOST                                      
      hints.ai_socktype = SOCK_STREAM
```
I deleted the `= 0` and ` = null` lines because I thought the single comment
on the `alloc` line was more parsimonious and less annoying (to me & to reviewers) that
two commented out lines with the same trailing comment: DRY.

For background: 
The C idiom goes something like (freestyle, not compiled):
```
   struct pollfd pfd = {fd, POLLIN, 0} // probably a runtime copy of zero

// or, less efficiently but often seen...
// requires knowing C structure initialization rules in order to figure out contents of revents.
   struct pollfd pfd = {0} // initialize all fields of structure to 0, contents of padding unspecified
   pfd.fd = fd
   pfd.events = POLLIN
```

Reply:
I think it is better to just leave the initializations as they were and add no comments. The changes are related but not needed for this PR. 

First, the Java specification guarantees zero initialization but only for heap objects.
https://docs.oracle.com/javase/specs/jls/se8/html/jls-4.html#jls-4.12.5

Second, I am not sure why we even have local / stack allocated memory zeroed. The Java compiler errors if a local variable is unset.
```java
int i;
System.out.println(i);
```
```java
Main.java:5: error: variable i might not have been initialized
    System.out.println(i);
                       ^
1 error
```
https://stackoverflow.com/questions/415687/why-are-local-variables-not-initialized-in-java

I think we should re-evaluate why we zero stack allocated memory. If it is not for Java or Scala compatibility I don't think we should do it especially as it decreases performance.